client.go

```diff
@@ -234,7 +236,15 @@ func (c *Client) ExecCmd(cmd *Cmd) ([]string, error) {
 		return nil, ErrNotConnected
 	}
 
-	c.work <- cmd.String()
+	c.mutex.Lock()
```

Comment:
I'm in too minds if we want to do this, it adds the lock overhead to a client which can only perform one request at a time. This falls under the remit of nothing being concurrent safe unless documented to be so.

Reply:
I can try to think of a better solution to solve this but it should definitely be concurrent safe imho.
Is the overhead really that significant in this use case?

Reply:
My issue doesn't come from trying to use the library from multiple goroutines at once.

My issue happens once unhandled timeout/connection closed errors occur.  (for example: ` read: connection reset by peer`)

Because the next time I execute a command on that closed connection, the `conn.Write()` call freezes indefinitely, causing the issue.

Reply:
enabling keepalive on the connection should fix this

Reply:
`conn.Write()` should also never freeze "indefinitely", only as long as the Timeout that is specified/`DefaultTimout`, since [there's a Deadline set](https://github.com/multiplay/go-ts3/blob/master/client.go#L218) before calling it.